src/Symfony/Bundle/FrameworkBundle/Command/BuildDebugContainerTrait.php

```diff
@@ -53,6 +53,9 @@ protected function getContainerBuilder(KernelInterface $kernel): ContainerBuilde
             (new XmlFileLoader($container = new ContainerBuilder(), new FileLocator()))->load($kernel->getContainer()->getParameter('debug.container.dump'));
             $locatorPass = new ServiceLocatorTagPass();
             $locatorPass->process($container);
+
+            $container->getCompilerPassConfig()->setBeforeOptimizationPasses([]);
+            $container->getCompilerPassConfig()->setOptimizationPasses([]);
```

Comment:
This is not necessary to fix the issue, but should prevent similar ones in the future.

Also, we might want to make `ContainerLintCommand` use this trait. I'd be happy to do it if it's ok for the scope of this PR.

Reply:
https://github.com/symfony/symfony/pull/45629#pullrequestreview-900538180